tests/migrations/test_state.py

```diff
@@ -870,6 +871,39 @@ class Meta:
         self.assertEqual([name for name, mgr in food_state.managers], ['food_mgr'])
         self.assertEqual(food_state.managers[0][1].args, ('a', 'b', 1, 2))
 
+    def test_metaclass_conflict(self):
+        """
+        Test making class with complex metadata inheritance
+        """
+        new_apps = Apps(['migrations'])
+
+        class DummyMetaClass(type):
+            pass
+
+        class Mixin(six.with_metaclass(DummyMetaClass, object)):
+            mixin_attr = True
+
+        class ItermidiateMetaClass(ModelBase, DummyMetaClass):
+            pass
+
+        # create the same Model without metaclassmaker as you would normally do
+        class MetaclassTestModel(six.with_metaclass(ItermidiateMetaClass, models.Model, Mixin)):
+            class Meta:
+                app_label = "migrations"
+                apps = new_apps
+                swappable = 'TEST_SWAPPABLE_MODEL'
+
+        # Reset apps to be able to render __fake__ modules
+        new_apps = Apps(['migrations'])
+        model_state = ModelState.from_model(MetaclassTestModel)
+
+        try:
+            class_obj = model_state.render(new_apps)
+        except TypeError:
+            raise AssertionError("Not resolved metaclass conflict")
```

Comment:
If you want to respect the user ability to choose their failureException ( https://github.com/python/cpython/blob/master/Lib/unittest/case.py#L357 ), you can use

```python
self.fail("Not resolved metaclass conflict")
```

instead of forcing the AssertionError.

Other thought : by doing this, you may lose all the details of the original ``TypeError``  exception which might prove useful for debugging when the test fail (but I'm less sure about this one since I'm not sure what might trigger the ``TypeError`` in the first place).